syntax = "proto3";

package NUgc;

message TFeedbackContext {
    message TRecord {
        uint64 Time = 1;
        string Value = 2;
    }

    message TObjectProps {
        map<string, TRecord> Props = 1;
    }

    message TTable {
        map<string, TObjectProps> Rows = 1;
    }

    message TData {
        map<string, TTable> Tables = 1;
    }

    message TPhotosRow {
        repeated string Photos = 1;
    }

    message TPhotosData {
        map<string, TPhotosRow> Objects = 1;
    }

    message TReviewInfo {
        uint64 Time = 1;
        string Text = 2;
    }

    message TReviewsList {
        repeated TReviewInfo Reviews = 1;
    }

    message TReviewsData {
        map<string, TReviewsList> Objects = 1;
    }

    string UserId = 1;
    string AppId = 2;
    string TableKeyName = 3;
    uint64 StartAt = 4;
    uint64 FinishAt = 5;

    TData Data = 6;
    TReviewsData ReviewsData = 7;
    TPhotosData PhotosData = 8;
}

